### DISCUSSION TITLE
Reason for FPS slowdown

### DISCUSSION THREAD

**Discussion by ghost** [community/community#139600] [Programming Help] [answered]
### Body

Is there anything that could cause a steady and consistent FPS drop in a three.js app?

The FPS in my three.js game, starts to dip at about 2 minutes in and goes down steady and slow until it reaches 0 (after about 30 minutes).

I first suspected a memory leak but chrome heap snapshots dont seem to be getting much bigger over time (see screenshot).

Is there anything else that could cause a steady consistent FPS drop like this?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by KhamessiTaha** [comment, marked as answer]
Try measuring app performance using the Performance tab in devtools. If it’s caused by JavaScript - it’ll tell you exactly what’s taking so much time.
